src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -144,12 +171,44 @@ public override void TransferFullDuplex(ReadOnlySpan<byte> dataToWrite, Span<byt
             }
         }
 
+        private void TransferWriteOnly(ReadOnlySpan<byte> dataToWrite)
+        {
+            int bitLen = _settings.DataBitLength;
+            int lastBit = bitLen - 1;
+
+            using (StartChipSelect())
+            {
+                for (int i = 0; i < dataToWrite.Length; i++)
+                {
+                    for (int j = 0; j < bitLen; j++)
+                    {
+                        using (StartBitTransfer())
+                        {
+                            int bit = _settings.DataFlow == DataFlow.MsbFirst ? lastBit - j : j;
+                            bool bitToWrite = ((dataToWrite[i] >> bit) & 1) == 1;
+                            WriteBit(bitToWrite);
+                        }
+                    }
+                }
+            }
+        }
+
         private bool ReadWriteBit(bool bitToWrite)
         {
+            if (_miso == -1)
+            {
+                throw new ArgumentException("Cannot read without a miso pin specified");
+            }
+
             _controller.Write(_mosi, bitToWrite);
             return (bool)_controller.Read(_miso);
         }
 
+        private void WriteBit(bool bitToWrite)
+        {
+            _controller.Write(_mosi, bitToWrite);
+        }
+
         /// <inheritdoc />
         public override void Read(Span<byte> data)
         {
```

Comment:
Read should always throw when MISO is -1

Reply:
I'm moving it up the call path instead so it doesn't have to checked for each bit.